### PR TITLE
[react-slider] Update types to support thumb index (v1.30)

### PR DESCRIPTION
**Pull request by danielgek**
Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Review by axelboc — APPROVED** [review]
Hey, thanks for that! 🚀 

While you're at it, could you please also add support for the new `ariaLabelledby` prop, as per https://github.com/zillow/react-slider/pull/217?

**Comment by loichuder** [comment]
> While you're at it, could you please also add support for the new `ariaLabelledby` prop, as per [zillow/react-slider#217](https://github.com/zillow/react-slider/pull/217)?

And update the version number in the first line of `types/react-slider/index.d.ts` ?

**Comment by danielgek** [comment]
> Hey, thanks for that! 
> 
> While you're at it, could you please also add support for the new `ariaLabelledby` prop, as per [zillow/react-slider#217](https://github.com/zillow/react-slider/pull/217)?



> > While you're at it, could you please also add support for the new `ariaLabelledby` prop, as per [zillow/react-slider#217](https://github.com/zillow/react-slider/pull/217)?
> 
> And update the version number in the first line of `types/react-slider/index.d.ts` ?


Both are done ;) thanks once again for the quick review

File: types/react-slider/index.d.ts [hunk 1]
```diff
@@ -25,6 +25,15 @@ export interface ReactSliderProps<T extends number | ReadonlyArray<number> = num
...
     ariaLabel?: T extends number ? string : ReadonlyArray<string>;
 
+
+    /**
+     * aria-labelledby for screen-readers to apply to the thumbs.
+     * Used when slider rendered with separate label.
+     * Use an array for more than one thumb.
+     * The length of the array must match the number of thumbs in the value array.
+     */
+    ariaLabelledby?: string | string[];
```
Comment (axelboc) [inline types/react-slider/index.d.ts:35]: It should probably be `T extends number ? string : ReadonlyArray<string>` like for `ariaLabel`

Comment (danielgek) [reply types/react-slider/index.d.ts]: fixed

**Comment by danielgek** [comment]
btw @loichuder we are not suposed to put the patch version there see this: https://github.com/Microsoft/dtslint/blob/master/docs/dt-header.md

**Comment by axelboc** [comment]
> 
> 
> btw @loichuder we are not suposed to put the patch version there see this: https://github.com/Microsoft/dtslint/blob/master/docs/dt-header.md

It should still be `1.3` as you had, though.

**Comment by danielgek** [comment]
Uhh you are right, its done ;)

**Review by loichuder — APPROVED** [review]
Thanks !

**Comment by danielgek** [comment]
@axelboc @loichuder btw the latest version on npm is 1.1.4 https://www.npmjs.com/package/react-slider i changed to 1.3 thats why build is failing(cant find that version on npm), should i revert to 1.1.4 ?

**Comment by danielgek** [comment]
reverted to 1.1.4 as its the only way build passes and seams right to me xD

**Comment by axelboc** [comment]
Hang on, 1.3 is right. It just means that zillow hasn't published the latest releases.

**Comment by danielgek** [comment]
ok changed back to 1.3, but he needs to release it before this goes in

**Comment by stonebk** [comment]
My bad! Was accidentally publishing to a private registry 🤦🏻 

Published now!

**Comment by danielgek** [comment]
@stonebk this should be good to go ;)

**Review by loichuder — APPROVED** [review]

**Comment by danielgek** [comment]
Ready to merge

**Comment by danielgek** [comment]
Thanks everyone ;)

**Pull request merged**